### PR TITLE
Increase screen brightness when QrCode is in full screen Mode (EXPOSUREAPP-8224)

### LINKED ISSUE #4606: Increase screen brightness to 100% when opening QR-codes
See https://github.com/corona-warn-app/cwa-wishlist/issues/571, as I could not transfer the issue from the wishlist repo to this repo.

---
Internal Tracking ID: [EXPOSUREAPP-8224](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8224)

### PR DESCRIPTION
**Pull request by rkw95**
Implemented setting screen brightness to 100% while a QR code is shown in full screen mode for better scanability. Does not apply to other views when a QR code is shown but not as full screen.

Drawback might be the necessity to give the cwa the permission to write system settings.

See #4606 for more information.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.3'
+        classpath 'com.android.tools.build:gradle:7.0.4'
```
Comment (mtwalli) [inline build.gradle:16]: Please revert this? we do the upgrade the version internally to make sure that  the latest version won't cause any issues for internal pipelines +. it is not related to the issue mentioned

File: Corona-Warn-App/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -25,6 +25,8 @@
 
     <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
 
+    <uses-permission android:name="android.permission.WRITE_SETTINGS" tools:ignore="ProtectedPermissions" />
+
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/AndroidManifest.xml:29]: this won't be necessary, please see my suggestion below in `QrCodeFullScreenFragment` file ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivity.kt [hunk 3]
```diff
@@ -86,6 +87,15 @@ class MainActivity : AppCompatActivity(), HasAndroidInjector {
...
 
+        // ask setting permission at runtime
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
+            val canWriteSettings = Settings.System.canWrite(this)
+            if (!canWriteSettings) {
+                val intent = Intent(Settings.ACTION_MANAGE_WRITE_SETTINGS)
+                startActivity(intent)
+            }
+        }
+
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivity.kt:98]: Please revert ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/qrcode/fullscreen/QrCodeFullScreenFragment.kt [hunk 4]
```diff
@@ -67,6 +83,16 @@ class QrCodeFullScreenFragment : Fragment(R.layout.fragment_qr_code_full_screen)
...
+        // Set screen brightness back to old brightness level
+        // Do this in a try catch block if user does not give permission to change system settings
+        if (Settings.System.canWrite(context)) {
+            Settings.System.putInt(
+                context?.contentResolver,
+                Settings.System.SCREEN_BRIGHTNESS,
+                this.brightness
+            )
+        }
     }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/qrcode/fullscreen/QrCodeFullScreenFragment.kt:96]: Please revert the changes in here and just try the following snippet ?
```kotlin 

    override fun onResume() {
        super.onResume()
        keepScreenOn(on = true)
    }

    override fun onPause() {
        super.onPause()
        keepScreenOn(on = false)
    }

    private fun keepScreenOn(on: Boolean) = with(requireActivity().window) {
        if (on) addFlags(FLAG_KEEP_SCREEN_ON) else clearFlags(FLAG_KEEP_SCREEN_ON)
        attributes = attributes.apply { screenBrightness = if (on) 1f else -1f }
    }

```

**Review by mtwalli — COMMENTED** [review]
Thanks! @rkw95  for your contribution.
Please see my suggestion below which does not require extra permissions

File: build.gradle (hunk 1 above)
Comment (rkw95) [reply build.gradle]: Done.

**Comment by rkw95** [comment]
> Thanks! @rkw95 for your contribution. Please see my suggestion below which does not require extra permissions

@mtwalli Thanks for your response! :) I will try to implement your suggestions.

**Comment by mtwalli** [comment]
@rkw95  if you don't have the time to address the comments, please let us know so we can take care of it ?

**Comment by mtwalli** [comment]
@rkw95  Closing in favor of https://github.com/corona-warn-app/cwa-app-android/pull/4624
